crates/fluvio-dataplane-protocol/src/batch.rs

```diff
@@ -740,8 +760,35 @@ mod test {
         assert_eq!(batch_created.last_offset_delta(), 2);
     }
 
+    #[test]
+    fn test_into_consumer_records_iter() {
+        let mut batch = Batch::from(vec![
+            Record::default(),
+            Record::default(),
+            Record::default(),
+        ]);
+
+        batch.header.first_timestamp = 1_500_000_000;
```

Comment:
shouldn't this be computed instead of hardcoded based on initialization?  Or this should be put under different tests for it?

Reply:
That's part of MemoryBatch that has all the clock related stuff